browser/ui/views/omnibox/brave_search_conversion_promotion_view.cc

```diff
@@ -282,11 +286,17 @@ void BraveSearchConversionPromotionView::SetTypeAndInput(
   }
 
   UpdateState();
+
+  brave_search_conversion::p3a::RecordOmniboxPromoShown(local_state_, type);
 }
 
 void BraveSearchConversionPromotionView::OnSelectionStateChanged(
     bool selected) {
   selected_ = selected;
+  if (selected) {
```

Comment:
|selected| means this promotion match is active one in the omnibox popup.
When one match is selected, it has thin (orange) vertical line at the left side of the match.
I assume you thought |selected| means user pressed enter key.
I'll check how to hook the enter key for this promotion match.

Reply:
Did you see any startup crash with this PR?
It seems happened very early during the startup and I can see it very frequently on my local Win10.
Current master or my PR(brave_search_conversion) branch doesn't have it.

Reply:
Thanks for clarifying the meaning of the `selected` boolean, and for pushing a fix for keyboard navigation. I actually did not consider or test navigating to the promo using the keyboard arrow keys + enter key.

Your fix worked for keyboard navigation and pressing the embedded button in the view, but not for clicking on the entire entry, outside of the button. I decided to override `OnURLOpenedFromOmnibox` instead of `OnInputAccepted` for this reason.